main.py

```diff
@@ -185,19 +193,25 @@ def process_images():
           latest = q.get_nowait()
         except Queue.Empty:
           break
-      recognizer.detect_and_show(latest)
+      try:
+        recognizer.detect_and_show(latest)
+      except cv2.error:
+        # Sometimes, cvtcolor fails.
```

Comment:
I think this only happens if the webcam/file failed to open -- should be unnecessary with the explicit check you added above.